In [3]:
# importowanie stats z modułu scipy - korzystasz z funkcji statystycznych dostępnych w bibliotece SciPy
from scipy import stats

# Zadanie 1 - Test dla wartości oczekiwanej pojedynczej prób

Zmierzono czas pracy (w godzinach) piętnastu wylosowanych bateryjek radiowych i
otrzymano następujące wyniki:

 35,34; 36,26; 30,54; 38,2; 37,59; 39,18; 33,16; 34,23;
27,9; 36,33; 32,39; 34,89; 35,7; 31,99; 34,03.


Zakładając, że czas pracy bateryjek ma rozkład normalny sprawdź na poziomie istotności 0,05
czy przeciętny czas pracy jest równy 35?


## 1. Sformułowanie hipotez


H0: μ=35 (przeciętny czas dzialania baterii jest rowny 35h)\
H1: μ≠35 (przeciętny czas dzialania baterii nie jest rowny 35h)


## 2. Wybór testu statystycznego
- Test parametryczny - wartość przeciętna
- Pojedyncza próba
- Skala ilorazowa

-> Test T

## 3. Poziom istotności  
α=0,05


## 4. Obliczenie statystyki testowej i p-value oraz podjęcie decyzji weryfikacyjnej

In [4]:
# Wagi netto próbki
czas = [ 35.34, 36.26, 30.54, 38.2, 37.59, 39.18, 33.16, 34.23, 27.9, 36.33, 32.39, 34.89, 35.7, 31.99, 34.03]

# Wartość oczekiwana
mu = 35

# Przeprowadź test t-studenta
statystyka_T, p_wartosc = stats.ttest_1samp(czas, mu)

# Wyświetl wyniki
print("Statystyka testowa:", statystyka_T)
print("P-wartość:", p_wartosc)

# Sprawdź istotność statystyczną na poziomie alfa = 0.05
alfa = 0.05
if p_wartosc < alfa:
    print("Odrzucamy hipotezę zerową - średni czas jest różny od 35 sekund.")
else:
    print("Nie ma podstaw do odrzucenia hipotezy zerowej - średni czas jest równy 35 sekund.")

Statystyka testowa: -0.625790453948021
P-wartość: 0.5415206438042331
Nie ma podstaw do odrzucenia hipotezy zerowej - średni czas jest równy 35 sekund.


Z otrzymanego wydruku możemy odczytać m.in.  wartość statystyki testowej (t = -0.625790453948021) oraz  p-wartość = 0.5415206438042331. \
Ponieważ: p>α, zatem nie ma podstaw do odrzucenia hipotezy zerowej. Oznacza to, że średni czas dzialania baterii nie rozni sie znacznie od 35h


## 5. Wnioski:
Oznacza to, że średni czas dzialania baterii nie rozni sie znacznie od 35h

Należy pamiętać, że można popełnić tu błąd II rodzaju polegający na nieodrzuceniu hipotezy zerowej, chociaż jest fałszywa.
